packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -939,14 +938,11 @@ export function areRenderContextsEqual(context1: GridRenderContext, context2: Gr
 export function computeOffsetLeft(
   columnPositions: number[],
   renderContext: GridColumnsRenderContext,
-  isRtl: boolean,
   pinnedLeftLength: number,
 ) {
-  const factor = isRtl ? -1 : 1;
```

Comment:
I don't fully understand the purpose of this offset, but removing this fixes this additional space between right pinned cells and non-pinned cells:

<img width="1115" alt="Screenshot 2024-09-11 at 16 41 29" src="https://github.com/user-attachments/assets/6972f8a3-f427-4631-8f2b-63c0bc399e36">